cgrok/tree.c

```diff
@@ -93,7 +93,6 @@ void tctreeput(TCTREE *tree, const void *kbuf, int ksiz, const void *vbuf, int v
   void ** valPtr = dict_insert(tree->dict, key, &inserted);
   if (!inserted) {
     free(key);
-    free(*valPtr);
```

Comment:
This was causing use-after-free errors in Valgrind. The TC docs are mum about whether an overwritten value is usable afterwards.